lib/daily.js

```diff
@@ -0,0 +1,135 @@
+var _ = require('lodash');
+var moment = require('moment');
+var Promise = require('bluebird');
+var debug = require('debug')('daily');
+var nconf = require('nconf');
+ 
+var mongo = require('./mongo');
+
+
+function byDayStats(req) {
+    /* it is clear this is not they way, but this is just legacy alpha */
+    var what = _.get(req.params, 'what');
+	var queryMap = {
+		'impressions': [
+			{ 
+				'name': 'impressions',
+				'column': nconf.get('schema').impressions,
+				'timevar': '$impressionTime',
+				'filter': {},
+				'aggext': {}
+			},
+			{
+				'name': 'htmls',
+				'column': nconf.get('schema').htmls,
+				'timevar': '$savingTime',
+				'filter': {},
+				'aggext': {}
+			},
+			{
+				'name': 'timelines',
+				'column': nconf.get('schema').timelines,
+				'timevar': '$startTime',
+				'filter': {},
+				'aggext': {}
+			}
+		],
+	};
+
+	var statsMap  = _.get(queryMap, what, null);
+
+	if(_.isNull(statsMap)) {
+    	debug("%s byDayStats invalid request %s", req.randomUnicode, what);
+		throw new Error("Invalid request to /daily/(.*) ");
+	}
+
+    debug("%s byDayStats %s", req.randomUnicode, what);
+
+	return dailyQuery(statsMap)
+		.then(function(result) {
+			return { "json": result };
+		});
+};
+
+function dailyQuery(queryMap) {
+    
+    return Promise.map(queryMap, function(qM) {
+        return mongo
+			.countByDay(qM.column, qM.timevar, qM.filter, qM.aggext);
+    })
+    .map(function(unnamed, i) {
+		return _.map(unnamed, function(dateObj) {
+			dateObj.name = queryMap[i].name;
+			return dateObj;
+		});
+	})
+	.then(function(layered) {
+		return _.flatten(layered);
+	})
+	.reduce(aggregateByDate, {})
+	.then(function(collection) {
+		return _.orderBy(collection, function(stOb) {
```

Comment:
the aggregation in the DB makes one entry per day, so will not be a problem also in the long term. but generally I agree in moving these operation in mongo functions, just, I've not yet learn how to do it properly :)